core/src/main/java/com/flipkart/poseidon/core/PoseidonRequest.java

```diff
@@ -24,20 +24,22 @@
 import java.util.Enumeration;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Optional;
 import java.util.concurrent.ConcurrentHashMap;
 
 public class PoseidonRequest implements Request {
 
     private final Map<String, Object> attributes = new ConcurrentHashMap<>();
     private final String url;
+    private final Optional<HttpServletRequest> httpRequestOpt;
     private final ImmutableMap<String, Cookie> cookies;
     private final ImmutableMap<String, String> headers;
 
     public PoseidonRequest(String url, ImmutableMap<String, Cookie> cookies, ImmutableMap<String, String> headers, Map<String, String[]> attributes) {
         this.url = url;
         this.cookies = cookies;
         this.headers = headers;
-
+        this.httpRequestOpt = null;
```

Comment:
minor comment on the naming convention, we don't need to suffix it with "Opt".

also use Optional.empty() instead of null